Objdump-Change

ms-xy opened the pull request:
This is an improvement and at the same time a slight setback.
- Improved information output regarding truncating after the configured limit
- Lower service speed (somewhere around 8x slower it seems as it now parses ALL lines not just up to the configured limit, this could be circumvented by analyzing the headers table first using "objdump -w -h /tmp/filehash" to retrieve them)

Signed-off-by: Maximilian Schott ms@roottec.com

ms-xy committed:
This is an improvement and at the same time a slight setback.

+ Improved information output regarding truncating at the configured limit
- Speed (somewhere around 8x slower it seems as it now parses ALL lines not just up to the configured limit, this could be circumvented by analyzing the headers table first using "objdump -w -h /tmp/filehash" to retrieve them)

Signed-off-by: Maximilian Schott <ms@roottec.com>

ms-xy commented:
Hmm resolved itself without intervention ... weird

(edit: deleted previous comment about travis having problems)

webstergd commented:
i forced a recheck.

webstergd commented:
Could you provide a time comparison between the original and this method. Also, could you please update the readme so it provides samples output. i.e. [gogadget's readme.md](https://github.com/HolmesProcessing/Holmes-Totem/blob/master/src/main/scala/org/holmesprocessing/totem/services/gogadget/README.md)

ms-xy committed:
Improved Readme and service itself.

+ Output example and explanation as well as adjusted Usage
+ Cleaned up unused structs / variables
+ Cleaned up handler_analyze
	+ Moved structs / constant variables out of it
	+ Added objdump header parsing to avoid parsing all opcodes
	  (speedup of 2.5s -> 750ms for a 1MB binary (~400k opcodes))
+ Improved line parsing (exported into a goroutine, effectively speeding up processing a lot)

ms-xy commented:
Ok, implemented a lot of improvements. It is now as fast as the original (no significant difference observable, both roughly ~750ms on e.g. a 1MB binary with 400k opcodes). Updated the Readme as well.
Changed the output to include section flags as given in the executable file's headers.
